test/wallet-http.js

```diff
@@ -0,0 +1,650 @@
+    cbAddress = (await wallet.createAddress('default')).address;
+    await wallet.createAccount(accountTwo);
+  });
+
+  after(async () => {
+    await nclient.close();
+    await wclient.close();
+    await node.close();
+  });
+
+  beforeEach(async () => {
+    name = await nclient.execute('grindname', [5]);
+  });
+
+  afterEach(async () => {
+    await node.mempool.reset();
+  });
+
+  it('should mine to the primary/default wallet', async () => {
+    const height = 20;
+
+    for (let i = 0; i < height; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    const info = await nclient.getInfo();
+    assert.equal(info.chain.height, height);
+  });
+
+  it('should mine to the secondary/default wallet', async () => {
+    const {address} = await wallet2.createAddress('default');
+    for (let i = 0; i < 5; i++)
+      await nclient.execute('generatetoaddress', [1, address]);
+
+    assert.ok(true);
+  });
+
+  it('should have no name state indexed', async () => {
+    const names = await wallet.getNames();
+
+    assert.equal(names.length, 0);
+  });
+
+  it('should create an open and broadcast the transaction', async () => {
+    const json = await wallet.createOpen({
+      name: name
+    });
+
+    let entered = false;
+    node.mempool.on('tx', () => entered = true);
+
+    await sleep(100);
+
+    assert.equal(entered, true);
+    const mempool = await nclient.getMempool();
+
+    assert.ok(mempool.includes(json.hash));
+
+    const mtx = MTX.fromJSON(json);
+
+    const opens = mtx.outputs.filter(output => output.covenant.type === types.OPEN);
+    assert.equal(opens.length, 1);
+  });
+
+  it('should create an open and not broadcast the transaction', async () => {
+    const json = await wallet.createOpen({
+      name: name,
+      broadcast: false
+    });
+
+    let entered = false;
+    node.mempool.on('tx', () => entered = true);
+
+    await sleep(100);
+
+    // tx is not in the mempool
+    assert.equal(entered, false);
+
+    const mtx = MTX.fromJSON(json);
+    assert.ok(mtx.hasWitness());
+
+    const tx = mtx.toTX();
+
+    // tx is valid
+    assert.ok(tx.verify(mtx.view));
+
+    const opens = tx.outputs.filter(output => output.covenant.type === types.OPEN);
+    assert.equal(opens.length, 1);
+  });
+
+  it('should create an open and not sign the transaction', async () => {
+    const json = await wallet.createOpen({
+      name: name,
+      broadcast: false,
+      sign: false
+    });
+
+    let entered = false;
+    node.mempool.on('tx', () => entered = true);
+
+    await sleep(100);
+
+    assert.equal(entered, false);
+
+    const mtx = MTX.fromJSON(json);
+
+    assert.equal(mtx.verify(), false);
+  });
+
+  it('should throw error with incompatible broadcast and sign options', async () => {
+    const fn = async () => await (wallet.createOpen({
+      name: name,
+      broadcast: true,
+      sign: false
+    }));
+
+    assert.rejects(fn, 'Must sign when broadcasting');
```

Comment:
Not sure why, but `assert.rejects()` doesn't seem to match the exact error message. I can tell because you are missing a period here :-) But any arbitrary string in this assertion will pass the test.

Reply:
Yeah you are right that any string here will allow the assertion to pass. I thought that it did a regex under the hood against the error message? I've definitely experienced it doing that before in the past, maybe we have a different version of `bmocha`?

Reply:
I think the string is the error message `assert` outputs if the first argument is falsey.